File: py/day1.py

def main():
	f = open("/home/detmer/programming/aoc/aoc_input/input_day1.txt", "r")
	expenses = []
	for line in f:
		expenses.append(int(line))

	for x in expenses:
		#print(x)
		for y in expenses:
			#print(y)
			if(x + y == 2020):
				print(x * y)
				return 0;



if __name__ == '__main__':
	main()

File: 2020/c/day1.c
#define _GNU_SOURCE
#include <stdio.h>
#include <stdlib.h>

int main(void){

	FILE* filePointer;
	int bufferLength = 255;
	char buffer[bufferLength];
	int expenses[200];
	int i = 0;

	filePointer = fopen("/home/detmer/programming/aoc/2020/aoc_input/input_day1.txt", "r");

	while(fgets(buffer, bufferLength, filePointer)) {
    	expenses[i] = atoi(buffer);
    	i++;
	}

	fclose(filePointer);

	for(int x = 0; x < 200; x++){
		int sum = expenses[x] + expenses[x+1];
		for(int y = 1; y < 200; y++){
			int sum = expenses[x] + expenses[y];
			if(sum == 2020){
				printf("%d\n", expenses[x] * expenses[y]);
				return 0;
			}
		}
	}
    
}